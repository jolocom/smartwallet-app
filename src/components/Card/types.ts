import { StyleProp, ViewStyle } from 'react-native'
import { ClaimEntry } from 'jolocom-lib/js/credentials/credential/types'

export interface IWithCustomStyle<T = ViewStyle> {
  customStyles?: StyleProp<T>
}

export interface IField {
  label: string
  value: ClaimEntry
}

export interface ICardProps {
  id: string
  type: string
  optionalFields: IField[]
  mandatoryFields: Array<IField | null>
  photo?: string | undefined
  highlight?: string | undefined
}

interface IOptionalFields extends IWithCustomStyle {
  lastFieldPadding?: string
}

export interface ICardComposition {
  OptionalFields: React.FC<IOptionalFields>
  DocumentHeader: React.FC
  OtherHeader: React.FC
  Highlight: React.FC
  Photo: React.FC
  Dots: React.FC<IWithCustomStyle>
}
