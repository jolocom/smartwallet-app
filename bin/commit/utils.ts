import childProcess from 'child_process'

export const promisify =
  <T>(fn: (args: T) => void) =>
  (args: T): Promise<void> =>
    new Promise((res, rej) => {
      try {
        fn(args)
        res()
      } catch (e) {
        rej(e)
      }
    })

export const listStagedFiles = () =>
  childProcess.execSync(`git diff --cached --name-only --diff-filter=ACMR`)

export const stageModifiedFiles = (stagedFiles: string) =>
  childProcess.execSync(`echo "${stagedFiles}" | xargs git add`)

export const abortScript = (msg: string) => {
  console.log('')
  console.log('\x1b[0;33m%s\x1b[0m', msg)
  console.log('\x1b[0;31m%s\x1b[0m', 'Aborting')
  process.exit(1)
}

export const logStep = (msg: string) => {
  console.log('\x1b[0;34m%s\x1b[0m', `${msg}`)
}

export const spawnProcess = (
  onClose: (
    code: number | null,
    dataOutput: string,
    errorOutput: string,
  ) => void,
  cmd: string,
  args?: readonly string[],
  options?: childProcess.SpawnOptionsWithoutStdio,
) => {
  const spawnedProcess = childProcess.spawn(cmd, args, options)

  let dataOutput = ''
  let errorOutput = ''

  spawnedProcess.stdout.on('data', (data) => {
    dataOutput += data.toString() + '\n'
  })
  spawnedProcess.stderr.on('data', (error: Buffer) => {
    errorOutput += error.toString() + '\n'
  })

  spawnedProcess.stderr.pipe(process.stderr)
  spawnedProcess.stdout.pipe(process.stdout)

  spawnedProcess.on('close', (code) => onClose(code, dataOutput, errorOutput))

  return spawnedProcess
}
