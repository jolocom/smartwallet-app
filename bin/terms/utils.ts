import fetch from 'node-fetch';
import fs from 'fs';
import shell from 'shelljs'

const CURRENT_PATH = process.cwd();
const SECRETS_REPO = 'dev@hetz1.jolocom.io:poeditor-key'
const BASE_URL = 'https://api.poeditor.com/v2';
const API_KEY_LOCATION = `${CURRENT_PATH}/bin/poeditor-key/POEDITOR_TOKEN.txt`
const PROJECT_ID_LOCATION = `${CURRENT_PATH}/bin/poeditor-key/POEDITOR_SW2_PROJECT_ID.txt`

export enum Languages {
  en = 'en',
  de = 'de',
}

interface IResult  {
  [key: string]: {
    [key: string]: number
  }
}

export interface IResponse<T = IResult> {
  response: {
    status: string,
    code: string,
    message: string,
  },
  result: T
}

export const cloneSecrets = () => {
  return new Promise((res, rej) => {
    const keyExists = fs.existsSync(API_KEY_LOCATION);
    const idExists = fs.existsSync(PROJECT_ID_LOCATION);
    if(!keyExists || !idExists) {
      shell.cd(`${CURRENT_PATH}/bin`)
      shell.rm('-rf', `poeditor-key`)
      shell.exec(
        `git clone ${SECRETS_REPO} -b master`,
        {},
        (code, stdout, stderr) => {
          if (code != 0) return rej(new Error(stderr))
          return res(stdout)
        },
      )
    } else {
      res('Secrets are already present');
    }
  })
}

const readTextFile = (location: string) => {
  return new Promise<string>((res, rej) => {
    fs.readFile(location, 'utf8', (err, data) => {
      if (err) rej(err)
      else res(data.toString().trim())
    })
  })
}

const getPoeditorSecrets = async () => {
    const apiKey = await readTextFile(API_KEY_LOCATION)
    const projectId = await readTextFile(PROJECT_ID_LOCATION)
    return { apiKey, projectId }
}


export const sendPostRequest = async <T = IResponse, >(path: string, params: Record<string, string>) => {
  try {
    const { apiKey, projectId } = await getPoeditorSecrets();
    const requestParams = new URLSearchParams({
      api_token: apiKey,
      id: projectId,
      ...params
    })
    const res = await fetch(`${BASE_URL}${path}`, {
      method: 'POST',
      body: requestParams.toString(),
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded'
      }
    })
    return await res.json() as T;
  } catch(err) {
    console.error(err);
  }
}