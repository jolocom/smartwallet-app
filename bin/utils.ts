import fs from 'fs'
import { stringify as stringifyEnv, parse as parseEnv, Data } from 'envfile'
import shell from 'shelljs'

export const cloneRepo = async (remote: string, repoName: string) => {
  return new Promise<string>((res, rej) => {
    const dir = shell.tempdir()
    console.log(`${dir}/${repoName}`)
    shell.rm('-rf', `${dir}/${repoName}`)
    shell.exec(
      `git clone ${remote}:${repoName} -b master ${dir}/${repoName}`,
      {},
      (code, stdout, stderr) => {
        if (code != 0) return rej(new Error(stderr))
        return res(`${dir}/${repoName}`)
      },
    )
  })
}

export const catFile = async (path: string) => {
  return new Promise((res, rej) => {
    const dsn = shell.cat(path)
    if (dsn.stderr) rej(dsn.stderr)
    res(dsn.stdout)
  })
}

export const writeEnv = async (path: string, data: object) => {
  const env = stringifyEnv(data)

  return new Promise((res, rej) => {
    fs.writeFile(path, env, (err) => {
      if (err) rej(err)
      else res(undefined)
    })
  })
}

export const readEnv = async (path: string) => {
  return new Promise<Data>((res, rej) => {
    fs.readFile(path, 'utf8', (err, data) => {
      if (err) {
        console.log(
          '\x1b[4;31m%s\x1b[0m',
          'Error! Could not find the environmental config (.env).',
        )
      } else res(parseEnv(data))
    })
  })
}
