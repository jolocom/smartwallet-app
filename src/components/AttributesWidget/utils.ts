import { AttrsState } from '~/modules/attributes/types'
import { SelectableAttrI } from './index'

export const getMappedAttrubited = (attrs: AttrsState<string>) =>
  Object.keys(attrs).reduce((acc, val) => {
    acc[val] = attrs[val].map((attr) => ({
      val: attr,
      isSelected: false,
    }))
    return acc
  }, {} as AttrsState<SelectableAttrI>)
