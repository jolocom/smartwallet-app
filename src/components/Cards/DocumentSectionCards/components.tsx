import { DisplayVal } from '@jolocom/sdk/js/credentials'
import React from 'react'
import {
  Image,
  ImageBackground,
  StyleProp,
  StyleSheet,
  TouchableOpacity,
  View,
  ViewStyle,
} from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import { TextLayoutEvent } from '~/types/props'

import { Colors } from '~/utils/colors'
import { debugView } from '~/utils/dev'
import { Fonts } from '~/utils/fonts'
import { useCredentialNameScale, usePruneFields } from '../hooks'
import { FieldsCalculator } from '../InteractionShare/components'
import { ScaledText, ScaledView } from '../ScaledCard'

interface Props {
  onPress: () => void
  positionStyles: Partial<Pick<ViewStyle, 'left' | 'right' | 'top' | 'bottom'>>
}

export const CardMoreBtn: React.FC<Props> = ({ onPress }) => (
  <ScaledView
    scaleStyle={styles.dotsContainerScaled}
    style={styles.dotsContainer}
  >
    <TouchableOpacity onPress={onPress} style={styles.dotsBtn}>
      {[...Array(3).keys()].map((c) => (
        <ScaledView key={c} scaleStyle={styles.dot} />
      ))}
    </TouchableOpacity>
  </ScaledView>
)

export const DocumentFooter: React.FC<{
  rightIcons?: string[]
  leftIcons?: string[]
  style?: StyleProp<ViewStyle>
}> = ({ rightIcons, leftIcons, style = {} }) => {
  //TODO use icons
  return (
    <ScaledView
      style={[styles.footerContainer, style]}
      scaleStyle={styles.footerContainerScaled}
    >
      <View style={styles.footerBorder} />
    </ScaledView>
  )
}

export const DocumentPhoto: React.FC<{ photo: string }> = ({ photo }) => (
  <ScaledView
    scaleStyle={styles.photoContainerScaled}
    style={styles.photoContainer}
  >
    <Image resizeMode="cover" style={styles.photo} source={{ uri: photo }} />
  </ScaledView>
)

export const DocumentHeader: React.FC<{
  name: string
  icon?: string
  onPressMenu: () => void
}> = ({ name, icon, onPressMenu }) => {
  const { handleCredentialNameTextLayout } = useCredentialNameScale()

  return (
    <View style={styles.headerContainer}>
      {icon && (
        <ScaledView
          scaleStyle={{
            width: 32,
            height: 32,
            marginRight: 10,
          }}
        >
          <Image
            resizeMode="cover"
            style={[styles.photo, { borderRadius: 4.2 }]}
            source={{ uri: icon }}
          />
        </ScaledView>
      )}
      <ScaledText
        // @ts-expect-error
        onTextLayout={handleCredentialNameTextLayout}
        numberOfLines={1}
        scaleStyle={styles.credentialName}
        style={[styles.mediumText, { flex: 1 }]}
      >
        {name}
      </ScaledText>
      <CardMoreBtn
        onPress={onPressMenu}
        positionStyles={{
          top: 18,
          right: 17,
        }}
      />
    </View>
  )
}

export const DocumentHolderName: React.FC<{
  name: string
  onLayout: (e: TextLayoutEvent) => void
}> = ({ name, onLayout }) => {
  return (
    <ScaledView
      scaleStyle={{
        paddingLeft: 24,
      }}
    >
      <ScaledText
        // @ts-expect-error
        onTextLayout={onLayout}
        numberOfLines={2}
        style={styles.mediumText}
        scaleStyle={styles.holderName}
      >
        {name}
      </ScaledText>
    </ScaledView>
  )
}

export const DocumentFields: React.FC<{
  fields: Required<DisplayVal>[]
  maxFields: number
  maxLines: number
}> = ({ fields, maxLines, maxFields }) => {
  const {
    displayedFields,
    handleFieldValueLayout,
    handleFieldValuesVisibility,
  } = usePruneFields(fields, maxFields, maxLines)

  return (
    <View style={{ paddingLeft: 24 }}>
      <FieldsCalculator cbFieldsVisibility={handleFieldValuesVisibility}>
        {displayedFields.map((f, idx) => (
          <React.Fragment key={f.key}>
            {idx !== 0 && <ScaledView scaleStyle={{ paddingBottom: 14 }} />}
            <ScaledText
              numberOfLines={1}
              style={[
                styles.regularText,
                {
                  width: '100%',
                },
              ]}
              scaleStyle={styles.fieldLabel}
            >
              {f.label.trim()}:
            </ScaledText>
            <ScaledView scaleStyle={{ paddingBottom: 6 }} />
            <ScaledText
              numberOfLines={2}
              //@ts-expect-error
              onTextLayout={(e: TextLayoutEvent) =>
                handleFieldValueLayout(e, idx)
              }
              scaleStyle={styles.fieldText}
              style={[
                styles.mediumText,
                {
                  width: '100%',
                },
              ]}
            >
              {f.value}
            </ScaledText>
          </React.Fragment>
        ))}
      </FieldsCalculator>
    </View>
  )
}

const BackgroundOpacity = () => (
  <LinearGradient
    colors={['rgba(255,255,255,0)', 'rgba(255,255,255,1)']}
    style={{ flex: 1 }}
  />
)

export const DocumentBackgroundImage: React.FC<{ image: string }> = ({
  image,
}) => (
  <ScaledView scaleStyle={{ height: 112 }} style={{ width: '100%' }}>
    <ImageBackground
      style={{ width: '100%', height: '100%' }}
      source={{ uri: image }}
    >
      <LinearGradient
        colors={['rgba(255,255,255,0)', 'rgba(255,255,255,1)']}
        style={{ flex: 1 }}
      />
    </ImageBackground>
  </ScaledView>
)

export const DocumentBackgroundColor: React.FC<{ color: string }> = ({
  color,
}) => (
  <View style={{ height: 84, width: '100%', backgroundColor: color }}>
    <LinearGradient
      colors={['rgba(255,255,255,0)', 'rgba(255,255,255,1)']}
      style={{ flex: 1 }}
    />
  </View>
)

const styles = StyleSheet.create({
  dotsContainerScaled: {
    paddingVertical: 3,
    width: 24,
    height: 24,
  },
  dotsContainer: {
    //position: 'absolute',
    //zIndex: 100,
    height: '100%',
  },
  dotsBtn: {
    alignItems: 'stretch',
    justifyContent: 'space-between',
    flexDirection: 'column',
    paddingHorizontal: 10,
    flex: 1,
  },
  dot: {
    width: 4,
    height: 4,
    borderRadius: 2,
    marginHorizontal: 2,
    backgroundColor: Colors.black,
  },
  credentialName: {
    fontSize: 22,
    lineHeight: 26,
    fontWeight: '500',
  },
  holderName: {
    fontSize: 24,
    lineHeight: 28,
  },
  photoContainer: {
    position: 'absolute',
    overflow: 'hidden',
    zIndex: 10,
  },
  photoContainerScaled: {
    bottom: 27,
    right: 14,
    width: 82,
    height: 82,
    borderRadius: 41,
  },
  headerContainer: {
    height: 68,
    width: '100%',
    alignItems: 'center',
    flexDirection: 'row',
    padding: 16,
  },
  photo: {
    width: '100%',
    height: '100%',
  },
  footerContainerScaled: {
    height: 60,
    paddingHorizontal: 20,
  },
  footerContainer: {
    width: '100%',
    zIndex: 9,
  },
  footerBorder: {
    width: '100%',
    borderTopWidth: 1,
    borderTopColor: '#D8D8D8',
  },
  regularText: {
    fontFamily: Fonts.Regular,
    fontSize: 22,
    color: Colors.black,
  },
  mediumText: {
    fontFamily: Fonts.Medium,
    color: Colors.black,
  },
  fieldLabel: {
    fontSize: 16,
    lineHeight: 16,
    color: Colors.slateGray,
  },
  fieldText: {
    fontSize: 20,
    lineHeight: 20,
    letterSpacing: 0.14,
  },
})
