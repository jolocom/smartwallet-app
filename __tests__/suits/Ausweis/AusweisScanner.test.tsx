import React from 'react'
import { useNavigation, useRoute } from '@react-navigation/native'
import nfcManager from 'react-native-nfc-manager'
import { useNetInfo } from '@react-native-community/netinfo'

import {
  getMockedDispatch,
  mockSelectorReturn,
} from '../../mocks/libs/react-redux'
import { AusweisScannerState } from '~/screens/LoggedIn/eID/types'
import { renderWithSafeArea } from '../../utils/renderWithSafeArea'
import { AusweisScanner } from '~/screens/LoggedIn/eID/components'
import { useGoBack } from '~/hooks/navigation'
import * as eIDHooks from '~/screens/LoggedIn/eID/hooks'
import { fireEvent, waitFor } from '@testing-library/react-native'

jest.mock('@react-navigation/native')
jest.mock('../../../src/hooks/navigation')

describe('Ausweis scanner screen', () => {
  const mockedNavigate = jest.fn()
  const mockGoBack = jest.fn()
  const mockHandleDone = jest.fn()
  const mockHandleDismiss = jest.fn()
  const mockCheckNfcSupport = jest.fn()
  const mockDispatchFn = getMockedDispatch()
  beforeAll(() => {
    ;(nfcManager.isSupported as jest.Mock).mockReturnValue(true)
    ;(nfcManager.isEnabled as jest.Mock).mockReturnValue(true)
    ;(useNetInfo as jest.Mock).mockReturnValue({
      isConnected: true,
    })
    mockSelectorReturn({
      toasts: {
        active: null,
      },
      ausweis: {
        scannerKey: null,
      },
    })
    ;(useNavigation as jest.Mock).mockReturnValue({
      navigate: mockedNavigate,
    })
    ;(useRoute as jest.Mock).mockReturnValue({
      params: {
        onDone: mockHandleDone,
        state: AusweisScannerState.idle,
        onDismiss: mockHandleDismiss,
      },
    })
    ;(useGoBack as jest.Mock).mockReturnValue(mockGoBack)
    jest.spyOn(eIDHooks, 'useCheckNFC').mockReturnValue({
      checkNfcSupport: mockCheckNfcSupport,
    })
  })
  afterEach(() => {
    mockDispatchFn.mockClear()
    mockCheckNfcSupport.mockClear()
    mockHandleDismiss.mockClear()
    mockHandleDone.mockClear()
    mockGoBack.mockClear()
  })
  afterAll(() => {
    jest.resetAllMocks()
  })
  test('is displayed according to the designs', () => {
    const { toJSON } = renderWithSafeArea(<AusweisScanner />)
    expect(toJSON()).toMatchSnapshot()
  })
  test('scanner key is being dispatch to the store', () => {
    renderWithSafeArea(<AusweisScanner />)
    expect(mockDispatchFn).toHaveBeenCalled()
  })
  test('user can proceed when success animation has finished', async () => {
    ;(useRoute as jest.Mock).mockReturnValue({
      params: {
        onDone: mockHandleDone,
        state: AusweisScannerState.success,
        onDismiss: mockHandleDismiss,
      },
    })
    renderWithSafeArea(<AusweisScanner />)
    await waitFor(() => {
      expect(mockGoBack).toHaveBeenCalledTimes(1)
      expect(mockHandleDone).toHaveBeenCalledTimes(1)
    })
  })
  test('user can proceed when failure has occured', async () => {
    ;(useRoute as jest.Mock).mockReturnValue({
      params: {
        onDone: mockHandleDone,
        state: AusweisScannerState.failure,
        onDismiss: mockHandleDismiss,
      },
    })
    renderWithSafeArea(<AusweisScanner />)
    await waitFor(() => {
      expect(mockHandleDone).toHaveBeenCalledTimes(1)
    })
  })
  test('user can cancel scanning', async () => {
    ;(useRoute as jest.Mock).mockReturnValue({
      params: {
        onDone: mockHandleDone,
        state: AusweisScannerState.idle,
        onDismiss: mockHandleDismiss,
      },
    })
    const { getByText } = renderWithSafeArea(<AusweisScanner />)
    const cancelBtn = getByText('AusweisScanner.cancelBtn')
    fireEvent.press(cancelBtn)
    await waitFor(() => {
      expect(mockGoBack).toHaveBeenCalledTimes(1)
      expect(mockHandleDismiss).toHaveBeenCalledTimes(1)
    })
  })
})
