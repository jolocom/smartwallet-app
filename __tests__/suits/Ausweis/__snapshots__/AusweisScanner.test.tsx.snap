// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ausweis scanner screen is displayed according to the designs 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.65)",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#1f1f1f",
            "borderRadius": 20,
            "justifyContent": "center",
            "marginBottom": 10,
            "paddingBottom": 22,
            "paddingHorizontal": 20,
            "paddingTop": 32,
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
            "width": "96%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingBottom": 22,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "paddingTop": undefined,
                  "textAlign": "center",
                },
                Object {
                  "color": "rgba(255,255,255,0.85)",
                  "fontFamily": "TTCommons-Medium",
                  "fontSize": 28,
                  "letterSpacing": 0,
                  "lineHeight": 32,
                },
                Object {
                  "marginBottom": 32,
                },
              ]
            }
          >
            AusweisScanner.header
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 156,
                "justifyContent": "center",
                "width": 156,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="100%"
                bbWidth="100%"
                focusable={false}
                height="100%"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={156}
                vbWidth={156}
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup>
                  <RNSVGGroup
                    fill={null}
                    fillRule={0}
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                      ]
                    }
                  >
                    <RNSVGEllipse
                      cx={77.805}
                      cy={78}
                      fill={-13158601}
                      propList={
                        Array [
                          "fill",
                          "stroke",
                          "strokeWidth",
                        ]
                      }
                      rx={76.805}
                      ry={77}
                      stroke={-1}
                      strokeWidth={2}
                    />
                    <RNSVGGroup
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          12.593,
                          30.982,
                        ]
                      }
                      propList={
                        Array [
                          "stroke",
                        ]
                      }
                      stroke={-1}
                    >
                      <RNSVGPath
                        d="M9.894 28.983C3.298 40.105 0 47.32 0 50.625c0 3.306.734 14.592 2.2 33.858l-.785 6.41 19.907 19.125"
                        fill={-14737633}
                        propList={
                          Array [
                            "fill",
                            "strokeWidth",
                            "strokeLinecap",
                          ]
                        }
                        strokeLinecap={2}
                        strokeWidth={1.579}
                      />
                      <RNSVGRect
                        fill={-16743169}
                        height={44.36}
                        propList={
                          Array [
                            "fill",
                            "strokeWidth",
                          ]
                        }
                        rx={4.212}
                        strokeWidth={1.579}
                        width={75.415}
                        x={10.084}
                        y={0.79}
                      />
                      <RNSVGPath
                        d="M9.82 66.672c1.643-4.573 2.683-7.733 3.12-9.477 1.969-7.87-.446-8.425 1.783-14.743 1.475-4.184 6.19-14.977 6.685-15.27 3.106-1.835 6.078 3.92 6.24 9.478.072 2.459-1.678 7.214-2.229 12.637-.431 4.25 1.337 10.38 1.337 21.588 0 6.845-3.377 6.923-5.348 13.163-.407 1.289-.407 3.57 0 6.845l-.05 19.094"
                        fill={-14737633}
                        propList={
                          Array [
                            "fill",
                            "strokeWidth",
                            "strokeLinecap",
                          ]
                        }
                        strokeLinecap={2}
                        strokeWidth={1.579}
                      />
                      <RNSVGPath
                        d="M60.917 11.717c3.502 3.915 5.253 8.074 5.253 12.478 0 4.403-1.751 8.318-5.253 11.743"
                        propList={
                          Array [
                            "strokeWidth",
                            "strokeLinecap",
                          ]
                        }
                        strokeLinecap={2}
                        strokeWidth={1.8}
                      />
                      <RNSVGPath
                        d="M58.834 16.983c2.089 2.227 3.134 4.557 3.134 6.99 0 2.44-1.05 4.673-3.152 6.7m-2.393-9.629c1.192 1.028 1.789 2.103 1.789 3.226 0 1.126-.6 2.157-1.8 3.092"
                        propList={
                          Array [
                            "strokeWidth",
                            "strokeLinecap",
                          ]
                        }
                        strokeLinecap={2}
                        strokeWidth={1.8}
                      />
                    </RNSVGGroup>
                    <RNSVGGroup
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          69.7,
                          22.724,
                        ]
                      }
                    >
                      <RNSVGPath
                        d="M54.706 57.713c5.403 6.23 9.456 11.24 12.157 15.032 2.702 3.791 1.112 15.87-4.769 36.236l-.094.732c-3.907 4.17-6.91 7.023-9.008 8.563-2.098 1.54-5.257 3.206-9.476 5"
                        fill={-14737633}
                        propList={
                          Array [
                            "fill",
                            "stroke",
                            "strokeWidth",
                            "strokeLinecap",
                          ]
                        }
                        stroke={-1}
                        strokeLinecap={2}
                        strokeWidth={1.579}
                      />
                      <RNSVGPath
                        d="M23.128 7.35a6.552 6.552 0 013.062-3.997 6.558 6.558 0 014.993-.66l31.562 8.47a6.572 6.572 0 013.995 3.07 6.571 6.571 0 01.66 4.993L46.943 95.661a6.552 6.552 0 01-3.062 3.997 6.558 6.558 0 01-4.993.659l-31.562-8.47a6.572 6.572 0 01-3.995-3.07 6.571 6.571 0 01-.661-4.993z"
                        fill={-14737633}
                        propList={
                          Array [
                            "fill",
                            "stroke",
                            "strokeWidth",
                          ]
                        }
                        stroke={-1}
                        strokeWidth={1.579}
                      />
                      <RNSVGPath
                        d="M57.626 88.704c-1.718-4.58-2.806-7.745-3.262-9.493-2.058-7.882.466-8.438-1.864-14.766-1.543-4.191-6.47-15-6.99-15.294-3.246-1.838-6.353 3.927-6.523 9.493-.075 2.462 1.754 7.225 2.33 12.657.452 4.257-1.398 10.397-1.398 21.622 0 6.856 3.53 6.934 5.592 13.184.426 1.291.426 3.577 0 6.856l-1.746 10.563"
                        fill={-14737633}
                        propList={
                          Array [
                            "fill",
                            "stroke",
                            "strokeWidth",
                            "strokeLinecap",
                          ]
                        }
                        stroke={-1}
                        strokeLinecap={2}
                        strokeWidth={1.579}
                      />
                      <RNSVGEllipse
                        cx={24.96}
                        cy={87.649}
                        fill={-14737633}
                        propList={
                          Array [
                            "fill",
                            "stroke",
                          ]
                        }
                        rx={4.241}
                        ry={4.246}
                        stroke={-1}
                      />
                      <RNSVGPath
                        d="M40.877 9.418l11.142 3.199c.559.16.882.743.722 1.302l-.086.302a1.052 1.052 0 01-1.302.722L40.21 11.744a1.055 1.055 0 01-.722-1.303l.086-.301a1.052 1.052 0 011.303-.722z"
                        fill={-1}
                        propList={
                          Array [
                            "fill",
                          ]
                        }
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 74,
                    "justifyContent": "center",
                    "width": 74,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "borderColor": "#ffffff",
                      "borderRadius": 9,
                      "borderWidth": 1,
                      "height": 18,
                      "opacity": 1.3333333333333333,
                      "overflow": "hidden",
                      "position": "absolute",
                      "transform": Array [
                        Object {
                          "scale": 1,
                        },
                      ],
                      "width": 18,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "borderColor": "#ffffff",
                      "borderRadius": 9,
                      "borderWidth": 1,
                      "height": 18,
                      "opacity": 1.6666666666666665,
                      "overflow": "hidden",
                      "position": "absolute",
                      "transform": Array [
                        Object {
                          "scale": 0,
                        },
                      ],
                      "width": 18,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 74,
                    "justifyContent": "center",
                    "width": 74,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="100%"
                  bbWidth="100%"
                  focusable={false}
                  height="100%"
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": "100%",
                        "width": "100%",
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={32}
                  width="100%"
                >
                  <RNSVGGroup>
                    <RNSVGPath
                      d="M31.333.357a1.219 1.219 0 00-1.724 0L9.852 20.114 2.08 12.343a1.219 1.219 0 10-1.724 1.724L8.99 22.7a1.22 1.22 0 001.724 0L31.333 2.08a1.219 1.219 0 000-1.723z"
                      fill={-1}
                      fillRule={1}
                      propList={
                        Array [
                          "fill",
                          "fillRule",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 74,
                    "justifyContent": "center",
                    "width": 74,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="100%"
                  bbWidth="100%"
                  focusable={false}
                  height="100%"
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": "100%",
                        "width": "100%",
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                  width="100%"
                >
                  <RNSVGGroup>
                    <RNSVGPath
                      d="M10.852 12.004L.23 22.7a.764.764 0 000 1.076.75.75 0 001.068 0L12 13l10.703 10.778a.75.75 0 001.069 0 .764.764 0 000-1.076L13.148 12.004l10.63-10.706a.764.764 0 000-1.075.752.752 0 00-1.067 0L12 11.008 1.289.223a.752.752 0 00-1.068 0 .764.764 0 000 1.076l10.631 10.705z"
                      fill={-1}
                      fillRule={1}
                      propList={
                        Array [
                          "fill",
                          "fillRule",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "paddingTop": undefined,
                  "textAlign": "center",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "TTCommons-Regular",
                  "fontSize": 20,
                  "letterSpacing": 0.14,
                  "lineHeight": 22,
                },
                Object {
                  "marginTop": 36,
                  "paddingHorizontal": 10,
                },
              ]
            }
          >
            AusweisScanner.startSubheader
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginVertical": 5,
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="button"
        >
          <View
            style={
              Array [
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 8,
                    "justifyContent": "center",
                    "width": "100%",
                  },
                  false,
                  Object {
                    "marginVertical": 5,
                  },
                ],
                Object {
                  "backgroundColor": "rgba(81, 81, 81, 0.18)",
                },
                undefined,
                Object {
                  "height": 45,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "paddingTop": undefined,
                    "textAlignVertical": "center",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "TTCommons-Medium",
                    "opacity": 0.8,
                  },
                  Object {
                    "fontSize": 18,
                  },
                  Object {},
                ]
              }
            >
              AusweisScanner.cancelBtn
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
